packages/propeller/src/StreamsSink.mts

```diff
@@ -98,25 +120,31 @@ type CreateOptions = {
 }
 
 export class StreamsSink implements Sink {
-  private queue = new AsyncQueue<Stream>()
   private streams = new Map<StreamName, Stream>()
-  private activeStreams = new Set<StreamName>()
   private batchStreams = new Map<IngesterBatch, Set<Stream>>()
-  private readonly tracingAttrs: Attributes = {}
+  private events = new EventEmitter()
+  private readonly handler: EventHandler<string>
+  private dispatcher!: ConcurrentDispatcher
+  private tracingAttrs: Attributes
   constructor(
-    private readonly handle: EventHandler<string>,
-    private maxConcurrentStreams: number,
+    handler: EventHandler<string>,
+    private readonly maxConcurrentStreams: number,
     private readonly batchLimiter: BatchLimiter,
+    tracingAttrs: Attributes = {},
   ) {
-    this.addTracingAttrs({ "eqx.max_concurrent_streams": maxConcurrentStreams })
+    this.tracingAttrs = Object.assign({ "eqx.max_read_ahead": maxConcurrentStreams }, tracingAttrs)
+    this.handler = traceHandler(tracer, this.tracingAttrs, handler)
   }
 
   addTracingAttrs(attrs: Attributes) {
     Object.assign(this.tracingAttrs, attrs)
   }
 
+  onError(handler: (err: Error) => void) {
+    this.events.on("error", handler)
+  }
+
   private handleStreamCompletion(stream: Stream) {
```

Comment:
maybe handleStreamCompleted to convey that it only reacting and not directly handling
alternately name it removeStreamFromBatchesQueue or something that says what it does
(also wondering whether you have a risky semantic where one stream that continually has new items coming in can hijack the overall objective of getting the head batch completed asap.
While this may not be a problem with the current capabilities/use cases, eventually you might be merging all events into the stream buffers the minute they come into memory, but the scheduler's work should be driven solely by what the batches dictate)